src/transformers/models/rag/modeling_rag.py

```diff
@@ -556,7 +556,9 @@ def forward(
         if encoder_outputs is None:
 
             if has_to_retrieve:
-                question_enc_outputs = self.question_encoder(input_ids, attention_mask=attention_mask)
+                question_enc_outputs = self.question_encoder(
+                    input_ids, attention_mask=attention_mask, return_dict=True
```

Comment:
Do we have to do this ? I know that modeling_rag.py is currently not compatible with tuples, but it would actually be better to implement this functionality with tuples rather than forcing the ModelOutput....

Reply:
Do you think that `question_enc_outputs = self.question_encoder(...)[0]` would be better then ? (taking the embeddingss from the tuple)

Reply:
Actually, thinking more about since it's not possible to use tuples anyways, doing `return_dict=True` is fine (I do it as well in `generate()`....). Think we can just wait until someone posts an issue about not being able to use tuples with RAG. All good as it is for me - Sorry!

Reply:
I'm pretty sure we can use tuples actually. Why wouldn't it work ?

Reply:
I would work! It'll just be a horrible `if-else cascade` to know which position to take for which cases. If you look at T5 you can see that there are already quite a lot of ugly if else statements to retrieve the correct position. In Rag it'll be much worse